src/Chainweb/BlockHeader.hs

```diff
@@ -341,16 +351,15 @@ powTarget
         -- ^ block creation time of new block
     -> HashTarget
         -- ^ POW target of new block
-powTarget p (BlockCreationTime bt) = case window ver of
+powTarget p (BlockCreationTime bt) = case effectiveWindow p of
     Nothing -> maxTarget
-    Just (WindowWidth w)
-        | isLastInEpoche p ->
-            adjust ver (windowSize w) (t .-. _blockEpochStart p) (_blockTarget p)
+    Just w
+        | isLastInEpoch p ->
+            adjust ver w (t .-. _blockEpochStart p) (_blockTarget p)
```

Comment:
Ah, it seems like `adjust` is occurring at the correct point anyway. So is the issue is just with the naming of the function `isLastInEpoch`. Really it should be `isFirstInEpoch`.

Reply:
Are you sure that this is correct? `p` is the parent header. The current header is the first in the epoch if the parent is the last in the previous epoch.

Reply:
Oh, in that case, I was right the first time, and DA needs to occur when **the parent was the last in the epoch** and **the parent's `BlockHeight` is 1 less than the "mod point" (11, 23, etc).**